docs/kernels-overview.md

```diff
@@ -1,38 +1,44 @@
 # .NET Interactive Architectural Overview 
 
-The kernel concept in .NET Interactive is a component that accepts commands and produces outputs. The commands are  typically blocks of arbitrary code, and the outputs are events that describe the results and effects of that code. The `Kernel` class represents this core abstraction.
+The kernel concept in .NET Interactive is a component that accepts commands and produces outputs. The commands are typically blocks of code, and the outputs are events that describe the results and effects of that code, including executing code and providing language services such as completions and diagnostics. The `Kernel` class represents this core abstraction.
 
-A kernel doesn't have to run in its own process. The default `dotnet-interactive` configuration runs several kernels in one process, enabling scenarios such as language-switching and .NET variable sharing. But one or more kernels can also run out-of-process, which will be transparent from the point of view of someone using it.
+A kernel doesn't have to run in its own process. The default `dotnet-interactive` configuration runs several kernels in one process, enabling scenarios such as language-switching and .NET variable sharing by reference between C#, F#, and PowerShell. But one or more kernels can also run out-of-process, which will be transparent from the point of view of someone using it. The `Kernel` APIs are message-based and provide the same API for all kernels whether they are in-process, out-of-process, or running on remote machines.
```

Comment:
It seems that language-switching is a feature in Polyglot Notebook, but I don't see a reference to it. I had to google it in order to understand it. Is this a feature ?

Reply:
It's described in the marketplace page: https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode. The source for this is duplicated in two files, both updated in this PR:

* src/polyglot-notebooks-vscode/README.md
* src/polyglot-notebooks-vscode-insiders/README.md

I wonder if we should make this more prominent and/or add it to the repo's root README.